### ISSUE TITLE
rpm2archive doesn't check command line switches

### ISSUE THREAD

**Issue by pmatilai** [rpm-software-management/rpm#2210]
rpm2archive doesn't seem to have any checking for legitimate command line switches:

This simply hangs (waiting for stdin)
> rpm2archive --nosuch

As does this:
> rpm2archive --nosuch /tmp/telnet-0.17-86.fc36.x86_64.rpm 

This merrily completes:
> cat /tmp/telnet-0.17-86.fc36.x86_64.rpm |rpm2archive --nosuch > /tmp/foo

Discovered when I thought the newly added option was --nocompress although it was --nocompression, caused some headscratching.

**Comment by JetXujing** [comment]
Can we add processing similar to rpmkeys: Check whether the option is supported, if not, print related information and exit. 
I'd be happy to contribute if it's available.
```
[root@localhost ~]# rpmkeys --nosuch
rpmkeys: --nosuch: unknown option
```

**Comment by ffesti** [comment]
Resolved by 5e22e0c5833fb53f9c54652072316998ed0365e4
